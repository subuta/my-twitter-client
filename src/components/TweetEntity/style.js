import createWithStyles from 'src/utils/style'

import {
  merge,
  apply,
  variants,
  screen
} from 'css-as-js'

const TweetMedia = apply('py-2')

const TweetImage = apply('mt-2 border rounded')

export default createWithStyles({
  TweetImage,
  TweetMedia
})
