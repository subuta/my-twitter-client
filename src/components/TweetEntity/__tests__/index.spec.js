import React from 'react'
import { mount, shallow } from 'enzyme'
import sinon from 'sinon'
import { create } from 'react-test-renderer'

import TweetEntity from '../index'

// SEE: https://github.com/CookPete/react-player/blob/488f82da421ac9a27c2e52137767d19cc7f47a3e/test/players/FilePlayer.js#L12-L18
class MockMediaStream {}

global.navigator = {}
global.window = {
  MediaStream: MockMediaStream,
  URL: {
    createObjectURL: url => 'mockObjectURL'
  }
}

const ENTITIES = {
  ANIMATED_GIF: {
    hashtags: [],
    symbols: [],
    user_mentions: [],
    urls: [],
    media: [
      {
        id: 1049606350135685100,
        id_str: '1049606350135685120',
        indices: [
          12,
          35
        ],
        media_url: 'http://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg',
        media_url_https: 'https://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg',
        url: 'https://t.co/IIMBa4I4Vp',
        display_url: 'pic.twitter.com/IIMBa4I4Vp',
        expanded_url: 'https://twitter.com/subuta_nico/status/1049606383195152384/photo/1',
        type: 'photo',
        sizes: {
          large: {
            w: 220,
            h: 172,
            resize: 'fit'
          },
          thumb: {
            w: 150,
            h: 150,
            resize: 'crop'
          },
          medium: {
            w: 220,
            h: 172,
            resize: 'fit'
          },
          small: {
            w: 220,
            h: 172,
            resize: 'fit'
          }
        }
      }
    ]
  },

  PHOTO: {
    urls: [],
    media: [{
      id_str: '967319472796180480',
      type: 'photo',
      url: 'https://t.co/uc9iteuAQ8',
      display_url: 'pic.twitter.com/uc9iteuAQ8',
      media_url: 'http://pbs.twimg.com/media/DWyb73hVoAA4RKr.jpg',
      sizes: {
        small: {
          w: 680,
          h: 477,
          resize: 'fit'
        },
        large: {
          w: 1838,
          h: 1290,
          resize: 'fit'
        }
      },
      video_info: null,
      additional_media_info: null
    }]
  }
}

const EXTENDED_ENTITIES = {
  ANIMATED_GIF: {
    media: [
      {
        id: 1049606350135685100,
        id_str: '1049606350135685120',
        indices: [
          12,
          35
        ],
        media_url: 'http://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg',
        media_url_https: 'https://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg',
        url: 'https://t.co/IIMBa4I4Vp',
        display_url: 'pic.twitter.com/IIMBa4I4Vp',
        expanded_url: 'https://twitter.com/subuta_nico/status/1049606383195152384/photo/1',
        type: 'animated_gif',
        sizes: {
          large: {
            w: 220,
            h: 172,
            resize: 'fit'
          },
          thumb: {
            w: 150,
            h: 150,
            resize: 'crop'
          },
          medium: {
            w: 220,
            h: 172,
            resize: 'fit'
          },
          small: {
            w: 220,
            h: 172,
            resize: 'fit'
          }
        },
        video_info: {
          aspect_ratio: [
            55,
            43
          ],
          variants: [
            {
              bitrate: 0,
              content_type: 'video/mp4',
              url: 'https://video.twimg.com/tweet_video/DpDzahXU8AAb81_.mp4'
            }
          ]
        }
      }
    ]
  },

  PHOTO: {
    urls: [],
    media: [
      {
        id_str: '967319472796180480',
        type: 'photo',
        url: 'https://t.co/uc9iteuAQ8',
        display_url: 'pic.twitter.com/uc9iteuAQ8',
        media_url: 'http://pbs.twimg.com/media/DWyb73hVoAA4RKr.jpg',
        sizes: {
          small: { w: 680, h: 477, resize: 'fit' },
          large: { w: 1838, h: 1290, resize: 'fit' }
        },
        video_info: null,
        additional_media_info: null
      }, {
        id_str: '967319508791656448',
        type: 'photo',
        url: 'https://t.co/uc9iteuAQ8',
        display_url: 'pic.twitter.com/uc9iteuAQ8',
        media_url: 'http://pbs.twimg.com/media/DWyb99nVAAAD2pC.jpg',
        sizes: {
          small: { w: 680, h: 477, resize: 'fit' },
          large: { w: 1839, h: 1291, resize: 'fit' }
        },
        video_info: null,
        additional_media_info: null
      }, {
        id_str: '967319543302402048',
        type: 'photo',
        url: 'https://t.co/uc9iteuAQ8',
        display_url: 'pic.twitter.com/uc9iteuAQ8',
        media_url: 'http://pbs.twimg.com/media/DWyb_-LVMAA25Nb.jpg',
        sizes: {
          small: { w: 680, h: 477, resize: 'fit' },
          large: { w: 1839, h: 1291, resize: 'fit' }
        },
        video_info: null,
        additional_media_info: null
      }, {
        id_str: '967319660986183680',
        type: 'photo',
        url: 'https://t.co/uc9iteuAQ8',
        display_url: 'pic.twitter.com/uc9iteuAQ8',
        media_url: 'http://pbs.twimg.com/media/DWycG0lVMAA8u27.jpg',
        sizes: {
          small: { w: 680, h: 477, resize: 'fit' },
          large: { w: 1838, h: 1290, resize: 'fit' }
        },
        video_info: null,
        additional_media_info: null
      }]
  }
}

test('should render with empty props.', () => {
  const tree = mount(
    <TweetEntity
      entities={{}}
      extendedEntities={{}}
    />
  )

  // Should render nothing
  expect(tree).toMatchSnapshot()
})

test('should render with PHOTO.', () => {
  const tree = mount(
    <TweetEntity
      entities={ENTITIES.PHOTO}
    />
  )

  // Testing for snapshot.
  expect(tree).toMatchSnapshot()
})

test('should render with ANIMATED_GIF.', () => {
  const tree = mount(
    <TweetEntity
      entities={ENTITIES.ANIMATED_GIF}
      extendedEntities={EXTENDED_ENTITIES.ANIMATED_GIF}
    />
  )

  // Testing for snapshot.
  expect(tree).toMatchSnapshot()
})
