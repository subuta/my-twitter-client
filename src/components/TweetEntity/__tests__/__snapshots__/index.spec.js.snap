// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with ANIMATED_GIF. 1`] = `
<Styled(branch(branch(branch(Component))))
  entities={
    Object {
      "hashtags": Array [],
      "media": Array [
        Object {
          "display_url": "pic.twitter.com/IIMBa4I4Vp",
          "expanded_url": "https://twitter.com/subuta_nico/status/1049606383195152384/photo/1",
          "id": 1049606350135685100,
          "id_str": "1049606350135685120",
          "indices": Array [
            12,
            35,
          ],
          "media_url": "http://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
          "media_url_https": "https://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
          "sizes": Object {
            "large": Object {
              "h": 172,
              "resize": "fit",
              "w": 220,
            },
            "medium": Object {
              "h": 172,
              "resize": "fit",
              "w": 220,
            },
            "small": Object {
              "h": 172,
              "resize": "fit",
              "w": 220,
            },
            "thumb": Object {
              "h": 150,
              "resize": "crop",
              "w": 150,
            },
          },
          "type": "photo",
          "url": "https://t.co/IIMBa4I4Vp",
        },
      ],
      "symbols": Array [],
      "urls": Array [],
      "user_mentions": Array [],
    }
  }
  extendedEntities={
    Object {
      "media": Array [
        Object {
          "display_url": "pic.twitter.com/IIMBa4I4Vp",
          "expanded_url": "https://twitter.com/subuta_nico/status/1049606383195152384/photo/1",
          "id": 1049606350135685100,
          "id_str": "1049606350135685120",
          "indices": Array [
            12,
            35,
          ],
          "media_url": "http://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
          "media_url_https": "https://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
          "sizes": Object {
            "large": Object {
              "h": 172,
              "resize": "fit",
              "w": 220,
            },
            "medium": Object {
              "h": 172,
              "resize": "fit",
              "w": 220,
            },
            "small": Object {
              "h": 172,
              "resize": "fit",
              "w": 220,
            },
            "thumb": Object {
              "h": 150,
              "resize": "crop",
              "w": 150,
            },
          },
          "type": "animated_gif",
          "url": "https://t.co/IIMBa4I4Vp",
          "video_info": Object {
            "aspect_ratio": Array [
              55,
              43,
            ],
            "variants": Array [
              Object {
                "bitrate": 0,
                "content_type": "video/mp4",
                "url": "https://video.twimg.com/tweet_video/DpDzahXU8AAb81_.mp4",
              },
            ],
          },
        },
      ],
    }
  }
>
  <Style
    Component={[Function]}
    Style={
      FreeStyle {
        "_children": Object {},
        "_counters": Object {},
        "_keys": Array [],
        "changeId": 0,
        "changes": Object {
          "add": [Function],
          "change": [Function],
          "remove": [Function],
        },
        "debug": true,
        "hash": [Function],
        "id": "f2",
        "sheet": Array [],
      }
    }
    componentName="branch(branch(branch(Component)))"
    componentProps={
      Object {
        "entities": Object {
          "hashtags": Array [],
          "media": Array [
            Object {
              "display_url": "pic.twitter.com/IIMBa4I4Vp",
              "expanded_url": "https://twitter.com/subuta_nico/status/1049606383195152384/photo/1",
              "id": 1049606350135685100,
              "id_str": "1049606350135685120",
              "indices": Array [
                12,
                35,
              ],
              "media_url": "http://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
              "media_url_https": "https://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
              "sizes": Object {
                "large": Object {
                  "h": 172,
                  "resize": "fit",
                  "w": 220,
                },
                "medium": Object {
                  "h": 172,
                  "resize": "fit",
                  "w": 220,
                },
                "small": Object {
                  "h": 172,
                  "resize": "fit",
                  "w": 220,
                },
                "thumb": Object {
                  "h": 150,
                  "resize": "crop",
                  "w": 150,
                },
              },
              "type": "photo",
              "url": "https://t.co/IIMBa4I4Vp",
            },
          ],
          "symbols": Array [],
          "urls": Array [],
          "user_mentions": Array [],
        },
        "extendedEntities": Object {
          "media": Array [
            Object {
              "display_url": "pic.twitter.com/IIMBa4I4Vp",
              "expanded_url": "https://twitter.com/subuta_nico/status/1049606383195152384/photo/1",
              "id": 1049606350135685100,
              "id_str": "1049606350135685120",
              "indices": Array [
                12,
                35,
              ],
              "media_url": "http://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
              "media_url_https": "https://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
              "sizes": Object {
                "large": Object {
                  "h": 172,
                  "resize": "fit",
                  "w": 220,
                },
                "medium": Object {
                  "h": 172,
                  "resize": "fit",
                  "w": 220,
                },
                "small": Object {
                  "h": 172,
                  "resize": "fit",
                  "w": 220,
                },
                "thumb": Object {
                  "h": 150,
                  "resize": "crop",
                  "w": 150,
                },
              },
              "type": "animated_gif",
              "url": "https://t.co/IIMBa4I4Vp",
              "video_info": Object {
                "aspect_ratio": Array [
                  55,
                  43,
                ],
                "variants": Array [
                  Object {
                    "bitrate": 0,
                    "content_type": "video/mp4",
                    "url": "https://video.twimg.com/tweet_video/DpDzahXU8AAb81_.mp4",
                  },
                ],
              },
            },
          ],
        },
        "styles": Object {
          "TweetImage": "branch(branch(branch(Component)))_TweetImage_f45h",
          "TweetMedia": "branch(branch(branch(Component)))_TweetMedia_f45h",
        },
      }
    }
  >
    <branch(branch(branch(Component)))
      entities={
        Object {
          "hashtags": Array [],
          "media": Array [
            Object {
              "display_url": "pic.twitter.com/IIMBa4I4Vp",
              "expanded_url": "https://twitter.com/subuta_nico/status/1049606383195152384/photo/1",
              "id": 1049606350135685100,
              "id_str": "1049606350135685120",
              "indices": Array [
                12,
                35,
              ],
              "media_url": "http://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
              "media_url_https": "https://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
              "sizes": Object {
                "large": Object {
                  "h": 172,
                  "resize": "fit",
                  "w": 220,
                },
                "medium": Object {
                  "h": 172,
                  "resize": "fit",
                  "w": 220,
                },
                "small": Object {
                  "h": 172,
                  "resize": "fit",
                  "w": 220,
                },
                "thumb": Object {
                  "h": 150,
                  "resize": "crop",
                  "w": 150,
                },
              },
              "type": "photo",
              "url": "https://t.co/IIMBa4I4Vp",
            },
          ],
          "symbols": Array [],
          "urls": Array [],
          "user_mentions": Array [],
        }
      }
      extendedEntities={
        Object {
          "media": Array [
            Object {
              "display_url": "pic.twitter.com/IIMBa4I4Vp",
              "expanded_url": "https://twitter.com/subuta_nico/status/1049606383195152384/photo/1",
              "id": 1049606350135685100,
              "id_str": "1049606350135685120",
              "indices": Array [
                12,
                35,
              ],
              "media_url": "http://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
              "media_url_https": "https://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
              "sizes": Object {
                "large": Object {
                  "h": 172,
                  "resize": "fit",
                  "w": 220,
                },
                "medium": Object {
                  "h": 172,
                  "resize": "fit",
                  "w": 220,
                },
                "small": Object {
                  "h": 172,
                  "resize": "fit",
                  "w": 220,
                },
                "thumb": Object {
                  "h": 150,
                  "resize": "crop",
                  "w": 150,
                },
              },
              "type": "animated_gif",
              "url": "https://t.co/IIMBa4I4Vp",
              "video_info": Object {
                "aspect_ratio": Array [
                  55,
                  43,
                ],
                "variants": Array [
                  Object {
                    "bitrate": 0,
                    "content_type": "video/mp4",
                    "url": "https://video.twimg.com/tweet_video/DpDzahXU8AAb81_.mp4",
                  },
                ],
              },
            },
          ],
        }
      }
      styles={
        Object {
          "TweetImage": "branch(branch(branch(Component)))_TweetImage_f45h",
          "TweetMedia": "branch(branch(branch(Component)))_TweetMedia_f45h",
        }
      }
    >
      <branch(branch(Component))
        entities={
          Object {
            "hashtags": Array [],
            "media": Array [
              Object {
                "display_url": "pic.twitter.com/IIMBa4I4Vp",
                "expanded_url": "https://twitter.com/subuta_nico/status/1049606383195152384/photo/1",
                "id": 1049606350135685100,
                "id_str": "1049606350135685120",
                "indices": Array [
                  12,
                  35,
                ],
                "media_url": "http://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
                "media_url_https": "https://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
                "sizes": Object {
                  "large": Object {
                    "h": 172,
                    "resize": "fit",
                    "w": 220,
                  },
                  "medium": Object {
                    "h": 172,
                    "resize": "fit",
                    "w": 220,
                  },
                  "small": Object {
                    "h": 172,
                    "resize": "fit",
                    "w": 220,
                  },
                  "thumb": Object {
                    "h": 150,
                    "resize": "crop",
                    "w": 150,
                  },
                },
                "type": "photo",
                "url": "https://t.co/IIMBa4I4Vp",
              },
            ],
            "symbols": Array [],
            "urls": Array [],
            "user_mentions": Array [],
          }
        }
        extendedEntities={
          Object {
            "media": Array [
              Object {
                "display_url": "pic.twitter.com/IIMBa4I4Vp",
                "expanded_url": "https://twitter.com/subuta_nico/status/1049606383195152384/photo/1",
                "id": 1049606350135685100,
                "id_str": "1049606350135685120",
                "indices": Array [
                  12,
                  35,
                ],
                "media_url": "http://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
                "media_url_https": "https://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
                "sizes": Object {
                  "large": Object {
                    "h": 172,
                    "resize": "fit",
                    "w": 220,
                  },
                  "medium": Object {
                    "h": 172,
                    "resize": "fit",
                    "w": 220,
                  },
                  "small": Object {
                    "h": 172,
                    "resize": "fit",
                    "w": 220,
                  },
                  "thumb": Object {
                    "h": 150,
                    "resize": "crop",
                    "w": 150,
                  },
                },
                "type": "animated_gif",
                "url": "https://t.co/IIMBa4I4Vp",
                "video_info": Object {
                  "aspect_ratio": Array [
                    55,
                    43,
                  ],
                  "variants": Array [
                    Object {
                      "bitrate": 0,
                      "content_type": "video/mp4",
                      "url": "https://video.twimg.com/tweet_video/DpDzahXU8AAb81_.mp4",
                    },
                  ],
                },
              },
            ],
          }
        }
        styles={
          Object {
            "TweetImage": "branch(branch(branch(Component)))_TweetImage_f45h",
            "TweetMedia": "branch(branch(branch(Component)))_TweetMedia_f45h",
          }
        }
      >
        <renderComponent(Component)
          entities={
            Object {
              "hashtags": Array [],
              "media": Array [
                Object {
                  "display_url": "pic.twitter.com/IIMBa4I4Vp",
                  "expanded_url": "https://twitter.com/subuta_nico/status/1049606383195152384/photo/1",
                  "id": 1049606350135685100,
                  "id_str": "1049606350135685120",
                  "indices": Array [
                    12,
                    35,
                  ],
                  "media_url": "http://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
                  "media_url_https": "https://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
                  "sizes": Object {
                    "large": Object {
                      "h": 172,
                      "resize": "fit",
                      "w": 220,
                    },
                    "medium": Object {
                      "h": 172,
                      "resize": "fit",
                      "w": 220,
                    },
                    "small": Object {
                      "h": 172,
                      "resize": "fit",
                      "w": 220,
                    },
                    "thumb": Object {
                      "h": 150,
                      "resize": "crop",
                      "w": 150,
                    },
                  },
                  "type": "photo",
                  "url": "https://t.co/IIMBa4I4Vp",
                },
              ],
              "symbols": Array [],
              "urls": Array [],
              "user_mentions": Array [],
            }
          }
          extendedEntities={
            Object {
              "media": Array [
                Object {
                  "display_url": "pic.twitter.com/IIMBa4I4Vp",
                  "expanded_url": "https://twitter.com/subuta_nico/status/1049606383195152384/photo/1",
                  "id": 1049606350135685100,
                  "id_str": "1049606350135685120",
                  "indices": Array [
                    12,
                    35,
                  ],
                  "media_url": "http://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
                  "media_url_https": "https://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
                  "sizes": Object {
                    "large": Object {
                      "h": 172,
                      "resize": "fit",
                      "w": 220,
                    },
                    "medium": Object {
                      "h": 172,
                      "resize": "fit",
                      "w": 220,
                    },
                    "small": Object {
                      "h": 172,
                      "resize": "fit",
                      "w": 220,
                    },
                    "thumb": Object {
                      "h": 150,
                      "resize": "crop",
                      "w": 150,
                    },
                  },
                  "type": "animated_gif",
                  "url": "https://t.co/IIMBa4I4Vp",
                  "video_info": Object {
                    "aspect_ratio": Array [
                      55,
                      43,
                    ],
                    "variants": Array [
                      Object {
                        "bitrate": 0,
                        "content_type": "video/mp4",
                        "url": "https://video.twimg.com/tweet_video/DpDzahXU8AAb81_.mp4",
                      },
                    ],
                  },
                },
              ],
            }
          }
          styles={
            Object {
              "TweetImage": "branch(branch(branch(Component)))_TweetImage_f45h",
              "TweetMedia": "branch(branch(branch(Component)))_TweetMedia_f45h",
            }
          }
        >
          <Component
            entities={
              Object {
                "hashtags": Array [],
                "media": Array [
                  Object {
                    "display_url": "pic.twitter.com/IIMBa4I4Vp",
                    "expanded_url": "https://twitter.com/subuta_nico/status/1049606383195152384/photo/1",
                    "id": 1049606350135685100,
                    "id_str": "1049606350135685120",
                    "indices": Array [
                      12,
                      35,
                    ],
                    "media_url": "http://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
                    "media_url_https": "https://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
                    "sizes": Object {
                      "large": Object {
                        "h": 172,
                        "resize": "fit",
                        "w": 220,
                      },
                      "medium": Object {
                        "h": 172,
                        "resize": "fit",
                        "w": 220,
                      },
                      "small": Object {
                        "h": 172,
                        "resize": "fit",
                        "w": 220,
                      },
                      "thumb": Object {
                        "h": 150,
                        "resize": "crop",
                        "w": 150,
                      },
                    },
                    "type": "photo",
                    "url": "https://t.co/IIMBa4I4Vp",
                  },
                ],
                "symbols": Array [],
                "urls": Array [],
                "user_mentions": Array [],
              }
            }
            extendedEntities={
              Object {
                "media": Array [
                  Object {
                    "display_url": "pic.twitter.com/IIMBa4I4Vp",
                    "expanded_url": "https://twitter.com/subuta_nico/status/1049606383195152384/photo/1",
                    "id": 1049606350135685100,
                    "id_str": "1049606350135685120",
                    "indices": Array [
                      12,
                      35,
                    ],
                    "media_url": "http://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
                    "media_url_https": "https://pbs.twimg.com/tweet_video_thumb/DpDzahXU8AAb81_.jpg",
                    "sizes": Object {
                      "large": Object {
                        "h": 172,
                        "resize": "fit",
                        "w": 220,
                      },
                      "medium": Object {
                        "h": 172,
                        "resize": "fit",
                        "w": 220,
                      },
                      "small": Object {
                        "h": 172,
                        "resize": "fit",
                        "w": 220,
                      },
                      "thumb": Object {
                        "h": 150,
                        "resize": "crop",
                        "w": 150,
                      },
                    },
                    "type": "animated_gif",
                    "url": "https://t.co/IIMBa4I4Vp",
                    "video_info": Object {
                      "aspect_ratio": Array [
                        55,
                        43,
                      ],
                      "variants": Array [
                        Object {
                          "bitrate": 0,
                          "content_type": "video/mp4",
                          "url": "https://video.twimg.com/tweet_video/DpDzahXU8AAb81_.mp4",
                        },
                      ],
                    },
                  },
                ],
              }
            }
            styles={
              Object {
                "TweetImage": "branch(branch(branch(Component)))_TweetImage_f45h",
                "TweetMedia": "branch(branch(branch(Component)))_TweetMedia_f45h",
              }
            }
          >
            <withHandlers(Component)
              autoPlay={true}
              className="mt-2"
              loop={true}
              muted={true}
              sources={
                Array [
                  Object {
                    "src": "https://video.twimg.com/tweet_video/DpDzahXU8AAb81_.mp4",
                    "type": "video/mp4",
                  },
                ]
              }
              style={
                Object {
                  "height": "auto",
                  "maxHeight": 172,
                  "width": 220,
                }
              }
            >
              <Component
                autoPlay={true}
                className="mt-2"
                loop={true}
                muted={true}
                play={[Function]}
                setRef={[Function]}
                sources={
                  Array [
                    Object {
                      "src": "https://video.twimg.com/tweet_video/DpDzahXU8AAb81_.mp4",
                      "type": "video/mp4",
                    },
                  ]
                }
                style={
                  Object {
                    "height": "auto",
                    "maxHeight": 172,
                    "width": 220,
                  }
                }
              >
                <video
                  autoPlay={true}
                  className="mt-2"
                  loop={true}
                  muted={true}
                  preload="auto"
                  style={
                    Object {
                      "height": "auto",
                      "maxHeight": 172,
                      "width": 220,
                    }
                  }
                >
                  <source
                    key="https://video.twimg.com/tweet_video/DpDzahXU8AAb81_.mp4"
                    src="https://video.twimg.com/tweet_video/DpDzahXU8AAb81_.mp4"
                    type="video/mp4"
                  />
                  Your browser does not support the video tag.
                </video>
              </Component>
            </withHandlers(Component)>
          </Component>
        </renderComponent(Component)>
      </branch(branch(Component))>
    </branch(branch(branch(Component)))>
  </Style>
</Styled(branch(branch(branch(Component))))>
`;

exports[`should render with PHOTO. 1`] = `
<Styled(branch(branch(branch(Component))))
  entities={
    Object {
      "media": Array [
        Object {
          "additional_media_info": null,
          "display_url": "pic.twitter.com/uc9iteuAQ8",
          "id_str": "967319472796180480",
          "media_url": "http://pbs.twimg.com/media/DWyb73hVoAA4RKr.jpg",
          "sizes": Object {
            "large": Object {
              "h": 1290,
              "resize": "fit",
              "w": 1838,
            },
            "small": Object {
              "h": 477,
              "resize": "fit",
              "w": 680,
            },
          },
          "type": "photo",
          "url": "https://t.co/uc9iteuAQ8",
          "video_info": null,
        },
      ],
      "urls": Array [],
    }
  }
>
  <Style
    Component={[Function]}
    Style={
      FreeStyle {
        "_children": Object {},
        "_counters": Object {},
        "_keys": Array [],
        "changeId": 0,
        "changes": Object {
          "add": [Function],
          "change": [Function],
          "remove": [Function],
        },
        "debug": true,
        "hash": [Function],
        "id": "f2",
        "sheet": Array [],
      }
    }
    componentName="branch(branch(branch(Component)))"
    componentProps={
      Object {
        "entities": Object {
          "media": Array [
            Object {
              "additional_media_info": null,
              "display_url": "pic.twitter.com/uc9iteuAQ8",
              "id_str": "967319472796180480",
              "media_url": "http://pbs.twimg.com/media/DWyb73hVoAA4RKr.jpg",
              "sizes": Object {
                "large": Object {
                  "h": 1290,
                  "resize": "fit",
                  "w": 1838,
                },
                "small": Object {
                  "h": 477,
                  "resize": "fit",
                  "w": 680,
                },
              },
              "type": "photo",
              "url": "https://t.co/uc9iteuAQ8",
              "video_info": null,
            },
          ],
          "urls": Array [],
        },
        "styles": Object {
          "TweetImage": "branch(branch(branch(Component)))_TweetImage_f45h",
          "TweetMedia": "branch(branch(branch(Component)))_TweetMedia_f45h",
        },
      }
    }
  >
    <branch(branch(branch(Component)))
      entities={
        Object {
          "media": Array [
            Object {
              "additional_media_info": null,
              "display_url": "pic.twitter.com/uc9iteuAQ8",
              "id_str": "967319472796180480",
              "media_url": "http://pbs.twimg.com/media/DWyb73hVoAA4RKr.jpg",
              "sizes": Object {
                "large": Object {
                  "h": 1290,
                  "resize": "fit",
                  "w": 1838,
                },
                "small": Object {
                  "h": 477,
                  "resize": "fit",
                  "w": 680,
                },
              },
              "type": "photo",
              "url": "https://t.co/uc9iteuAQ8",
              "video_info": null,
            },
          ],
          "urls": Array [],
        }
      }
      styles={
        Object {
          "TweetImage": "branch(branch(branch(Component)))_TweetImage_f45h",
          "TweetMedia": "branch(branch(branch(Component)))_TweetMedia_f45h",
        }
      }
    >
      <branch(branch(Component))
        entities={
          Object {
            "media": Array [
              Object {
                "additional_media_info": null,
                "display_url": "pic.twitter.com/uc9iteuAQ8",
                "id_str": "967319472796180480",
                "media_url": "http://pbs.twimg.com/media/DWyb73hVoAA4RKr.jpg",
                "sizes": Object {
                  "large": Object {
                    "h": 1290,
                    "resize": "fit",
                    "w": 1838,
                  },
                  "small": Object {
                    "h": 477,
                    "resize": "fit",
                    "w": 680,
                  },
                },
                "type": "photo",
                "url": "https://t.co/uc9iteuAQ8",
                "video_info": null,
              },
            ],
            "urls": Array [],
          }
        }
        styles={
          Object {
            "TweetImage": "branch(branch(branch(Component)))_TweetImage_f45h",
            "TweetMedia": "branch(branch(branch(Component)))_TweetMedia_f45h",
          }
        }
      >
        <branch(Component)
          entities={
            Object {
              "media": Array [
                Object {
                  "additional_media_info": null,
                  "display_url": "pic.twitter.com/uc9iteuAQ8",
                  "id_str": "967319472796180480",
                  "media_url": "http://pbs.twimg.com/media/DWyb73hVoAA4RKr.jpg",
                  "sizes": Object {
                    "large": Object {
                      "h": 1290,
                      "resize": "fit",
                      "w": 1838,
                    },
                    "small": Object {
                      "h": 477,
                      "resize": "fit",
                      "w": 680,
                    },
                  },
                  "type": "photo",
                  "url": "https://t.co/uc9iteuAQ8",
                  "video_info": null,
                },
              ],
              "urls": Array [],
            }
          }
          styles={
            Object {
              "TweetImage": "branch(branch(branch(Component)))_TweetImage_f45h",
              "TweetMedia": "branch(branch(branch(Component)))_TweetMedia_f45h",
            }
          }
        >
          <Component
            entities={
              Object {
                "media": Array [
                  Object {
                    "additional_media_info": null,
                    "display_url": "pic.twitter.com/uc9iteuAQ8",
                    "id_str": "967319472796180480",
                    "media_url": "http://pbs.twimg.com/media/DWyb73hVoAA4RKr.jpg",
                    "sizes": Object {
                      "large": Object {
                        "h": 1290,
                        "resize": "fit",
                        "w": 1838,
                      },
                      "small": Object {
                        "h": 477,
                        "resize": "fit",
                        "w": 680,
                      },
                    },
                    "type": "photo",
                    "url": "https://t.co/uc9iteuAQ8",
                    "video_info": null,
                  },
                ],
                "urls": Array [],
              }
            }
            styles={
              Object {
                "TweetImage": "branch(branch(branch(Component)))_TweetImage_f45h",
                "TweetMedia": "branch(branch(branch(Component)))_TweetMedia_f45h",
              }
            }
          >
            <div>
              hoge
            </div>
          </Component>
        </branch(Component)>
      </branch(branch(Component))>
    </branch(branch(branch(Component)))>
  </Style>
</Styled(branch(branch(branch(Component))))>
`;

exports[`should render with empty props. 1`] = `
<Styled(branch(branch(branch(Component))))
  entities={Object {}}
  extendedEntities={Object {}}
>
  <Style
    Component={[Function]}
    Style={
      FreeStyle {
        "_children": Object {},
        "_counters": Object {},
        "_keys": Array [],
        "changeId": 0,
        "changes": Object {
          "add": [Function],
          "change": [Function],
          "remove": [Function],
        },
        "debug": true,
        "hash": [Function],
        "id": "f2",
        "sheet": Array [],
      }
    }
    componentName="branch(branch(branch(Component)))"
    componentProps={
      Object {
        "entities": Object {},
        "extendedEntities": Object {},
        "styles": Object {
          "TweetImage": "branch(branch(branch(Component)))_TweetImage_f45h",
          "TweetMedia": "branch(branch(branch(Component)))_TweetMedia_f45h",
        },
      }
    }
  >
    <branch(branch(branch(Component)))
      entities={Object {}}
      extendedEntities={Object {}}
      styles={
        Object {
          "TweetImage": "branch(branch(branch(Component)))_TweetImage_f45h",
          "TweetMedia": "branch(branch(branch(Component)))_TweetMedia_f45h",
        }
      }
    >
      <renderComponent(Component)
        entities={Object {}}
        extendedEntities={Object {}}
        styles={
          Object {
            "TweetImage": "branch(branch(branch(Component)))_TweetImage_f45h",
            "TweetMedia": "branch(branch(branch(Component)))_TweetMedia_f45h",
          }
        }
      >
        <Component
          entities={Object {}}
          extendedEntities={Object {}}
          styles={
            Object {
              "TweetImage": "branch(branch(branch(Component)))_TweetImage_f45h",
              "TweetMedia": "branch(branch(branch(Component)))_TweetMedia_f45h",
            }
          }
        />
      </renderComponent(Component)>
    </branch(branch(branch(Component)))>
  </Style>
</Styled(branch(branch(branch(Component))))>
`;
